x-pack/plugins/cases/server/client/utils.ts

```diff
@@ -313,48 +313,29 @@ export const constructQueryOptions = ({
   };
 };
 
-interface CompareArrays {
-  addedItems: string[];
-  deletedItems: string[];
+interface CompareArrays<T> {
+  addedItems: T[];
+  deletedItems: T[];
 }
-export const compareArrays = ({
-  originalValue,
-  updatedValue,
-}: {
-  originalValue: string[];
-  updatedValue: string[];
-}): CompareArrays => {
-  const result: CompareArrays = {
-    addedItems: [],
-    deletedItems: [],
-  };
-  originalValue.forEach((origVal) => {
-    if (!updatedValue.includes(origVal)) {
-      result.deletedItems = [...result.deletedItems, origVal];
-    }
-  });
-  updatedValue.forEach((updatedVal) => {
-    if (!originalValue.includes(updatedVal)) {
-      result.addedItems = [...result.addedItems, updatedVal];
-    }
-  });
-
-  return result;
-};
 
-export const isTwoArraysDifference = (
-  originalValue: unknown,
-  updatedValue: unknown
-): CompareArrays | null => {
+export const arraysDifference = <T>(
+  originalValue: T[] | undefined | null,
+  updatedValue: T[] | undefined | null
+): CompareArrays<T> | null => {
   if (
     originalValue != null &&
     updatedValue != null &&
     Array.isArray(updatedValue) &&
     Array.isArray(originalValue)
   ) {
-    const compObj = compareArrays({ originalValue, updatedValue });
-    if (compObj.addedItems.length > 0 || compObj.deletedItems.length > 0) {
-      return compObj;
+    const addedItems = differenceWith(updatedValue, originalValue, isEqual);
```

Comment:
@cnasikas should we use `deepEqual` instead of `lodash.isEqual` here? might be faster 🤷‍♂️

Reply:
Maybe we can use `fast-deep-equal`? Nevertheless, I think it is ok to use `lodash.isEqual` and optimize later if it is a problem.